go-apps/meep-loc-serv/server/loc-serv.go

```diff
@@ -311,6 +361,287 @@ func registerUser(userAddress string, event []UserEventType, subsIdStr string) {
+						addressPairs = append(addressPairs, pair)
+					}
+				}
+			}
+
+			for _, pair := range addressPairs {
+				refAddr := pair.addr1
+				monitoredAddr := pair.addr2
+
+				var distParam gisClient.TargetPoint
+				distParam.AssetName = monitoredAddr
+
+				distResp, _, err := gisAppClient.GeospatialDataApi.GetDistanceGeoDataByName(context.TODO(), refAddr, distParam)
+				if err != nil {
+					log.Error("Failed to communicate with gis engine: ", err)
+					return
+				}
+
+				distance := int32(distResp.Distance)
+
+				switch *distanceCheck.Subscription.Criteria {
+				case ALL_WITHIN_DISTANCE:
+					if float32(distance) < distanceCheck.Subscription.Distance {
+						returnAddr[monitoredAddr] = &distResp
+					} else {
+						skipThisSubscription = true
+						break
+					}
+				case ALL_BEYOND_DISTANCE:
+					if float32(distance) > distanceCheck.Subscription.Distance {
+						returnAddr[monitoredAddr] = &distResp
+					} else {
+						skipThisSubscription = true
+						break
+					}
+				case ANY_WITHIN_DISTANCE:
+					if float32(distance) < distanceCheck.Subscription.Distance {
+						returnAddr[monitoredAddr] = &distResp
+					}
+				case ANY_BEYOND_DISTANCE:
+					if float32(distance) > distanceCheck.Subscription.Distance {
+						returnAddr[monitoredAddr] = &distResp
+					}
+				default:
+				}
+			}
+			if skipThisSubscription {
+				continue
+			}
+			if len(returnAddr) > 0 {
+				subsIdStr := strconv.Itoa(subsId)
+
+				var distanceNotif SubscriptionNotification
+				distanceNotif.DistanceCriteria = distanceCheck.Subscription.Criteria
+				distanceNotif.IsFinalNotification = false
+				distanceNotif.Link = distanceCheck.Subscription.Link
+				var terminalLocationList []TerminalLocation
+				for terminalAddr, distanceInfo := range returnAddr {
+					var terminalLocation TerminalLocation
+					terminalLocation.Address = terminalAddr
+					var locationInfo LocationInfo
+					locationInfo.Latitude = nil
+					locationInfo.Latitude = append(locationInfo.Latitude, distanceInfo.Latitude)
+					locationInfo.Longitude = nil
+					locationInfo.Longitude = append(locationInfo.Longitude, distanceInfo.Longitude)
+					locationInfo.Shape = 2
+					seconds := time.Now().Unix()
+					var timestamp TimeStamp
+					timestamp.Seconds = int32(seconds)
+					locationInfo.Timestamp = &timestamp
+					terminalLocation.CurrentLocation = &locationInfo
+					retrievalStatus := RETRIEVED
+					terminalLocation.LocationRetrievalStatus = &retrievalStatus
+					terminalLocationList = append(terminalLocationList, terminalLocation)
+				}
+				distanceNotif.TerminalLocation = terminalLocationList
+				distanceNotif.CallbackData = distanceCheck.Subscription.ClientCorrelator
+				var inlineDistanceSubscriptionNotification InlineSubscriptionNotification
+				inlineDistanceSubscriptionNotification.SubscriptionNotification = &distanceNotif
+				sendSubscriptionNotification(distanceCheck.Subscription.CallbackReference.NotifyURL, inlineDistanceSubscriptionNotification)
```

Comment:
Should we run this in a goroutine to avoid remote client from blocking Location Service execution?
NOTE: Same applies for existing notifications.

Reply:
i agree with your comment and that applies to every notifications... the reason why we block and wait for the response is to keep the sequences of the messages. Our system is meant to allow people to understand the behaviors of MEC apps and helping them to design those, not to offer a high performance system... but if we want to go to better efficiency... we can certainly use a go routine call for those notifications. TBD

Reply:
I'm okay with leaving it as is. If we see performance degradations or longer processing delays then we can address.